package/MDAnalysis/analysis/hbonds/wbridge_analysis.py

```diff
@@ -537,283 +897,375 @@ def _update_water_selection(self):
+                        new_route = route[:]
+                        new_route.append(new_node)
+                        new_node = self._expand_timeseries(new_node,'sele1_sele2')[3][:2]
+                        traverse_water_network(graph, new_node, end, new_route, maxdepth, result)
+        for s1 in selection_1:
+            route = [s1, ]
+            next_mol = self._expand_timeseries(s1,'sele1_sele2')[3][:2]
+            traverse_water_network(water_pool, next_mol, selection_2, route[:], self.order, result)
+
+        self._network.append(result['start'])
+
+    def _traverse_water_network(self, graph, current, analysis_func=None, output=None, link_func=None, **kwargs):
+        '''
+        This function recursively traverses the water network self._network and finds the hydrogen bonds which connect
+        the current atom to the next atom. The newly found hydrogen bond will be appended to the hydrogen bonds
+        connecting the selection 1 to the current atom via link_func. When selection 2 is reached, the full list of
+        hydrogen bonds connecting the selection 1 to selection 2 will be fed into analysis_func, which will then modify
+        the output in place.
+        :param graph: The connection network describes the connection between the atoms in the water network.
+        :param current: The hydrogen bonds from selection 1 until now.
+        :param analysis_func: The analysis function which is called to analysis the hydrogen bonds.
+        :param output: where the result is stored.
+        :param link_func: The new hydrogen bonds will be appended to current.
+        :param kwargs: the keywords which are passed into the analysis_func.
+        :return:
+        '''
+        if link_func is None:
+            # If no link_func is provided, the default link_func will be used
+            link_func = self._full_link
+
+        if graph is None:
+            # if selection 2 is reached
+            if not analysis_func is None:
+                # the result is analysed by analysis_func which will change the output
+                analysis_func(current, output, self.u, **kwargs)
+        else:
+            # make sure no loop can occur
+            if len(current) <= self.order:
+                for node in graph:
+                    # the new hydrogen bond will be added to the existing bonds
+                    new = link_func(current, node)
+                    self._traverse_water_network(graph[node], new, analysis_func, output, link_func, **kwargs)
+
+    def _expand_index(self, index):
+        '''
+        Expand the index into (resname, resid, name).
+        '''
+        atom = self.u.atoms[index]
+        return (atom.resname, atom.resid, atom.name)
 
-            As the _timeseries to timeserie conversion will be deprecated in 1.0
-            this function will automatically lose its value.
-        """
+    def _expand_timeseries(self, entry, output_format=None):
```

Comment:
What does "old" mean? Make this precise with version information, e.g. "format for release up to 0.19.2"

Reply:
The explanation has been added.